torch/_jit_internal.py

```diff
@@ -159,12 +159,90 @@ def fn(*args, **kwargs):
     }
     return fn
 
+def get_class_attribute_dict(fn, frames_up=2):
+    """
+    Get the frame for class attribute definitions (the top level frame
+    containing methods and class variables defined outside methods)
+    """
+    # default to 2 frames up:
+    #   0 - get_class_attribute_dict
+    #   1 - some decorator
+    #   2 - the class definition
+    frame = inspect.currentframe()
+    i = 0
+    while i < frames_up:
+        frame = frame.f_back
+        i += 1
 
-def ignore(fn):
-    ignored_fns.add(fn)
+    # The locals of the class definition frame contain any class attributes
+    # (such as __constants__)
+    return frame.f_locals
+
+def export(fn):
+    """
+    This decorator indicates that a method is used as an entry point into a
+    ScriptModule. `forward` implicitly is used as an entry point, so it does
+    not need this decorator.
+
+    Methods are added to a ScriptModule as they are called in Python. If a
+    method is never called, it will not be included in the ScriptModule when
+    saving. This decorator explicitly marks that a method should be included
+    even if it is not called from Python.
+    """
+    class_dict = get_class_attribute_dict(fn, frames_up=2)
```

Comment:
I feel like this would be idiomatic if it just set a `__torchscript_export` property on `fn` and the class then searched its own `__dict__` for things marked export. Searching the frames for what you hope to be the class dict is not great. What happens when we add `@property` and then we chain `@export @property def`. It would break.